lib/methods/ifwapichecktask.cpp

```diff
@@ -498,20 +430,28 @@ void IfwApiCheckTask::ScriptFunc(const Checkable::Ptr& checkable, const CheckRes
 	auto& io (IoEngine::Get().GetIoContext());
 	auto strand (Shared<asio::io_context::strand>::Make(io));
 	Shared<asio::ssl::context>::Ptr ctx;
-	double start = Utility::GetTime();
+
+	cr->SetExecutionStart(Utility::GetTime());
+	cr->SetCommand(cmdLine);
 
 	try {
 		ctx = SetupSslContext(cert, key, ca, crl, DEFAULT_TLS_CIPHERS, DEFAULT_TLS_PROTOCOLMIN, DebugInfo());
 	} catch (const std::exception& ex) {
-		ReportIfwCheckResult(checkable, cmdLine, cr, ex.what(), start, Utility::GetTime());
+		cr->SetOutput(ex.what());
+		cr->SetExecutionEnd(Utility::GetTime());
+
+		resultHandler(cr);
 		return;
 	}
 
 	auto conn (Shared<AsioTlsStream>::Make(io, *ctx, expectedSan));
 
+	// Handler for the final check result that is used to asynchronously process within the global non-I/O thread pool.
+	auto handler ([cr, resultHandler = std::move(resultHandler)]() { resultHandler(cr); });
```

Comment:
I don't really see the need for this optimization, I'd just capture `resultHandler` below and use it in a regular lambda with `QueueAsyncCallback()` even if that copies something once more. Doing this here makes little sense to me, if anything, `resultHandler` should directly capture `cr`.

Reply:
That comment was mainly about removing `const Value& cmdLine, const ProcessResult&`. If having `cr` as a parameter instead of capturing it just leads to you adding a seconds lambda that wraps the first one with `cr` additionally captured, that makes no sense.

But as I said:

> I don't really see the need for this optimization, I'd just capture `resultHandler` below and use it in a regular lambda with `QueueAsyncCallback()` even if that copies something once more.

We're talking about copying maybe 100 bytes more or less per check where we also establish a TLS connection, generate and parse JSON, etc., so that additional copy is irrelevant in the big picture and I'd favor what results in simpler code.